lib/Entry/Pass.hs

```diff
@@ -0,0 +1,228 @@
+-- SPDX-FileCopyrightText: 2022 Serokell <https://serokell.io>
+--
+-- SPDX-License-Identifier: MPL-2.0
+
+module Entry.Pass
+  ( passTextPrism
+  ) where
+
+import Coffer.Path (entryPathAsPath, mkEntryPath)
+import Control.Lens
+import Control.Monad (guard)
+import Data.Either.Extra (eitherToMaybe)
+import Data.HashMap.Lazy qualified as HS
+import Data.Maybe
+import Data.Set qualified as S
+import Data.Text (Text)
+import Data.Text qualified as T
+import Data.Time (UTCTime(UTCTime, utctDay), secondsToDiffTime)
+import Data.Time.Calendar.OrdinalDate.Compat
+import Data.Time.Clock.Compat (UTCTime(utctDayTime))
+import Data.Time.Format.ISO8601
+import Data.Void
+import Entry qualified as E
+import Fmt
+import Text.Megaparsec
+import Text.Megaparsec.Char
+
+data PassKv = PassKv (Maybe Text) [(Text, Text)]
+
+data PassField =
+  PassField
+  { pfKey :: Text
+  , pfDateModified :: Text
+  , pfVisibility :: Text
+  , pfValue :: Text
+  }
+  deriving stock (Show)
+makeLensesWith abbreviatedFields ''PassField
+
+data PassEntry =
```

Comment:
Why are these intermediate data structures needed?

In order to write an `Entry` to pass, we're converting `Entry -> PassEntry -> PassKv -> Text`

Reply:
I plan to remedy that globally, in the `Prism` rework. I want to merge Vault's format and pass' format. They're both KV, just pass uses a flat structure, vault a nested one. Names are different but that can be generalized too.

Reply:
> I plan to remedy that globally, in the Prism rework

Can you please create an issue for that? Please detail what will be done.

> I want to merge Vault's format and pass' format. They're both KV, just pass uses a flat structure, vault a nested one. Names are different but that can be generalized too.

Can you expand a bit on this? 
If I'm understanding correctly, that would justify the need for `PassKv`, but I still don't see the need for the `PassEntry` intermediate data structure (I haven't looked at this part of the code in detail _yet_, so I might just be missing something).

Reply:
> If I'm understanding correctly, that would justify the need for PassKv, but I still don't see the need for the PassEntry intermediate data structure (I haven't looked at this part of the code in detail yet, so I might just be missing something).
`PassEntry` exists, because the code is ugly and confusing and I thought that if I was trying to convert from `a` to `Text` and stuff into `HashMap Text Text` the code would be completely uncomprehensible.
> Can you please create an issue for that? Please detail what will be done.
sure